test/e2e/test_usage_status.py

```diff
@@ -1,181 +1,264 @@
+        :param detail: Expected detail to be returned.
+        """
+        assert self._delete_response_usage_status.status_code == status_code
+        assert self._delete_response_usage_status.json()["detail"] == detail
 
-    response = test_client.delete(f"/v1/usage-statuses/{str(ObjectId())}")
 
-    assert response.status_code == 404
-    assert response.json()["detail"] == "Usage status not found"
+class TestDelete(DeleteDSL):
+    """Tests for deleting a usage status."""
 
+    def test_delete(self):
+        """Test deleting a usage status."""
+        usage_status_id = self.post_usage_status(USAGE_STATUS_POST_DATA_NEW)
+        self.delete_usage_status(usage_status_id)
+        self.check_delete_usage_status_success()
 
-def test_delete_usage_status_that_is_a_part_of_item(test_client):
-    """Test trying to delete a usage status that is a part of an Item"""
-    # pylint: disable=duplicate-code
-    response = test_client.post(
-        "/v1/catalogue-categories", json=CATALOGUE_CATEGORY_POST_DATA_LEAF_NO_PARENT_NO_PROPERTIES
-    )
-    catalogue_category = response.json()
+        self.get_usage_status(usage_status_id)
+        self.check_get_usage_status_failed_with_detail(404, "Usage status not found")
 
-    response = test_client.post("/v1/systems", json=SYSTEM_POST_A)
-    system_id = response.json()["id"]
+    def test_delete_when_part_of_item(self):
+        """Test deleting a usage status when it is part of an item."""
+        usage_status_id = self.post_usage_status(USAGE_STATUS_POST_DATA_NEW)
 
-    response = test_client.post("/v1/manufacturers", json=MANUFACTURER_POST_DATA_REQUIRED_VALUES_ONLY)
+        # pylint: disable=duplicate-code
+        response = self.test_client.post(
+            "/v1/catalogue-categories", json=CATALOGUE_CATEGORY_POST_DATA_LEAF_NO_PARENT_NO_PROPERTIES
+        )
+        catalogue_category = response.json()
+
+        response = self.test_client.post("/v1/systems", json=SYSTEM_POST_DATA_REQUIRED_VALUES_ONLY)
+        system_id = response.json()["id"]
 
-    manufacturer_id = response.json()["id"]
+        response = self.test_client.post("/v1/manufacturers", json=MANUFACTURER_POST_DATA_REQUIRED_VALUES_ONLY)
 
-    catalogue_item_post = {
-        **CATALOGUE_ITEM_DATA_REQUIRED_VALUES_ONLY,
-        "catalogue_category_id": catalogue_category["id"],
-        "manufacturer_id": manufacturer_id,
-        "properties": add_ids_to_properties(
-            catalogue_category["properties"],
-            [
-                {"name": "Property A", "value": 20},
-                {"name": "Property B", "value": False},
-                {"name": "Property C", "value": "20x15x10"},
-            ],
-        ),
-    }
+        manufacturer_id = response.json()["id"]
 
-    response = test_client.post("/v1/catalogue-items", json=catalogue_item_post)
-    catalogue_item_id = response.json()["id"]
+        catalogue_item_post = {
+            **CATALOGUE_ITEM_POST_A,
+            "properties": [],
```

Comment:
We can probably just use `CATALOGUE_ITEM_DATA_REQUIRED_VALUES_ONLY`?
```suggestion
            **CATALOGUE_ITEM_DATA_REQUIRED_VALUES_ONLY,
```